stream_alert_cli/test.py

```diff
@@ -48,338 +56,486 @@ class TestingSuppressFilter(logging.Filter):
-    total_tests = failed_tests + passed_tests
-
-    # Print a message indicating how many of the total tests passed
-    print '\n\n{}({}/{})\tTests Passed{}'.format(
-        COLOR_GREEN,
-        passed_tests,
-        total_tests,
-        COLOR_RESET
-    )
-
-    # Check if there were failed tests and report on them appropriately
-    if rules_fail_pass[0]:
-        color = COLOR_RED
-        # Print a message indicating how many of the total tests failed
-        print '{}({}/{})\tTests Failed'.format(color, failed_tests, total_tests)
-
-        # Iterate over the rule_name values in the failed list and report on them
-        for index, failure in enumerate(rules_fail_pass[0]):
-            if index == failed_tests-1:
-                # Change the color back so std out is not red
-                color = COLOR_RESET
-            print '\t({}/{}) test failed for rule: {} [{}]{}'.format(index+1, failed_tests,
-                                                                     failure[0], failure[1],
-                                                                     color)
-
-    # Check if there were any warnings and report on them
-    if rules_fail_pass[2]:
-        color = COLOR_YELLOW
-        warning_count = len(rules_fail_pass[2])
-        print '{}{} \tWarning{}'.format(color, warning_count, ('', 's')[warning_count > 1])
-
-        for index, failure in enumerate(rules_fail_pass[2]):
-            if index == warning_count-1:
-                # Change the color back so std out is not yellow
-                color = COLOR_RESET
-            print '\t({}/{}) {} [{}]{}'.format(index+1, warning_count, failure[1],
-                                               failure[0], color)
-
-def test_rule(rule_name, test_record, formatted_record):
-    """Feed formatted records into StreamAlert and check for alerts
-    Args:
-        rule_name: The rule name being tested
-        test_record: A single record to test
-        formatted_record: A properly formatted version of record for the service to be tested
-
-    Returns:
-        boolean indicating if this rule passed
-    """
-    event = {'Records': [formatted_record]}
-
-    trigger_count = test_record.get('trigger_count')
-    if trigger_count:
-        expected_alert_count = trigger_count
-    else:
-        expected_alert_count = (0, 1)[test_record['trigger']]
 
-    # Start mocked sns
-    BOTO_MOCKER_SNS.start()
-
-    # Create the topic used for the mocking of alert sending
-    boto3.client('sns', region_name='us-east-1').create_topic(Name='test_streamalerts')
+class RuleProcessorTester(object):
+    """Class to encapsulate testing the rule processor"""
+
+    def __init__(self, print_output):
+        super(self.__class__, self).__init__()
+        # Create the topic used for the mocking of alert sending
+        # This is used in stream_alert/rule_processor/sink.py to 'send' alerts
+        sns_client = boto3.client('sns', region_name='us-east-1')
```

Comment:
I think I'm lacking the context where the mocking setup to prevent this from running on real infra